tests/posix/common/src/pthread_attr.c

```diff
@@ -439,6 +439,114 @@ ZTEST(pthread_attr, test_pthread_attr_setstacksize)
 	}
 }
 
+ZTEST(pthread_attr, test_pthread_attr_getinheritsched)
+{
+	int inheritsched = BIOS_FOOD;
+
+	/* degenerate cases */
+	{
+		if (false) {
+			/* undefined behaviour */
+			zassert_equal(pthread_attr_getinheritsched(NULL, NULL), EINVAL);
+			zassert_equal(pthread_attr_getinheritsched(NULL, &inheritsched), EINVAL);
+			zassert_equal(pthread_attr_getinheritsched(&uninit_attr, &inheritsched),
+				      EINVAL);
+		}
+		zassert_equal(pthread_attr_getinheritsched(&attr, NULL), EINVAL);
+	}
+
+	zassert_ok(pthread_attr_getinheritsched(&attr, &inheritsched));
+	zassert_equal(inheritsched, PTHREAD_INHERIT_SCHED);
+}
+
+static struct sched_param setinheritsched_sched_param;
+static int setinheritsched_policy;
+static int setinheritsched_inheritsched;
```

Comment:
@moonlight83340 - would it make sense to simply pass the appropriate `pthread_attr_t *` via `arg` in `test_pthread_attr_set_inheritsched_child_fn()`?